api/v1/connection_types.go

```diff
@@ -5,16 +5,232 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type ConnectionTelegram struct {
+	Token types.EnvVar `json:"token"`
+
+	// Chat ID or Channel name (using @channel-name)
+	Chats types.EnvVar `json:"chats"`
+}
+
+type ConnectionSlack struct {
+	Token types.EnvVar `json:"token"`
+
+	// Channel to send messages to in Cxxxxxxxxxx format
+	Channel string `json:"channel"`
+
+	BotName string `json:"botname"`
+}
+
+type ConnectionNtfy struct {
+	Topic string `json:"topic"`
+
+	Host     string       `json:"host"`
+	Username types.EnvVar `json:"username"`
+	Password types.EnvVar `json:"password"`
+}
+
+type ConnectionDiscord struct {
+	Token     string `json:"token"`
+	WebhookID string `json:"webhookID"`
+}
+
+type ConnectionEmail struct {
+	Host     string       `json:"host"`
+	Username types.EnvVar `json:"username"`
+	Password types.EnvVar `json:"password"`
+
+	// Default: false
+	InsecureTLS bool `json:"insecureTLS,omitempty"`
+
+	// Encryption Method
+	// 	Defulat: auto
+	// 	Possible values: None, ExplicitTLS, ImplicitTLS, Auto
+	Encryption string `json:"encryption,omitempty"`
+
+	// SMTP server port
+	// 	Default: 587
+	Port int `json:"port,omitempty"`
+
+	// Email address that the mail are sent from
+	FromAddress string `json:"fromAddress"`
+
+	// List of recipient e-mails
+	ToAddresses []string `json:"toAddresses"`
+
+	// The subject of the sent mail
+	Subject string `json:"subject"`
+
+	// Auth - SMTP authentication method
+	// Possible values: None, Plain, CRAMMD5, Unknown, OAuth2
+	Auth string `json:"auth"`
+}
+
+type ConnectionPushbullet struct {
+	Token types.EnvVar `json:"token"`
+
+	Targets []string `json:"targets"`
+}
+
+type ConnectionPushover struct {
+	Token types.EnvVar `json:"token"`
+	// User key
+	User string `json:"user"`
+}
+
+type ConnectionPostgres struct {
+	Host     types.EnvVar `json:"host"`
+	Username types.EnvVar `json:"username"`
+	Password types.EnvVar `json:"password"`
+}
+
+type ConnectionMySQL struct {
+	Host     types.EnvVar `json:"host"`
```

Comment:
All the SQL connections need to support specify a host + database and it builds the URL, or specifying just the URL